src/Signal-Estimator-GUI/ipc.cpp

```diff
@@ -0,0 +1,99 @@
+#include "ipc.h"
+
+QVector<QString> getOutputDevices(){
+    char buffer[128];
+    QVector<QString> Strvec;
+    // run pipe to get output devices from aplay
+    std::unique_ptr<FILE,decltype(&pclose)> pipe1(popen("aplay -l","r"),pclose);
+
+    if(!pipe1){
+        return Strvec;
+    }
+
+    // read everything from aplay into result
+    while (fgets(buffer,128,pipe1.get()) != nullptr){
```

Comment:
It would be better to use sizeof(buffer) instead of 128 here to avoid unnoticed errors if we change buffer size.

Reply:
done

Reply:
The same for getInputDevices().